src/libcore/num/flt2dec/mod.rs

```diff
@@ -410,8 +410,8 @@ fn determine_sign(sign: Sign, decoded: &FullDecoded, negative: bool) -> &'static
 /// it will only print given digits and nothing else.
 ///
 /// The byte buffer should be at least `MAX_SIG_DIGITS` bytes long.
-/// There should be at least 5 parts available, due to the worst case like
-/// `[+][0.][0000][45][0000]` with `frac_digits = 10`.
+/// There should be at least 4 parts available, due to the worst case like
+/// `[+][0.][0000][2][0000]` with `frac_digits = 10`.
```

Comment:
It's unfortunate that the `[]` notation is being used in two ways here: to denote a `Part` and to--I think--indicate the optional quality of the sign of the formatted number.  Since the `[+]` is being used in the latter sense, it does not count as a `Part` that would be stored in the `parts` array.